src/action.c

```diff
@@ -748,6 +754,37 @@ actions_run(struct view *activator, struct server *server,
 				view_shrink_to_edge(view, edge);
 			}
 			break;
+		case ACTION_TYPE_DIRECTIONAL_TARGET_WINDOW:
+			if (view) {
+				int dx, dy, distance;
+				enum view_edge direction = action_get_int(action, "direction", 0);
+				struct view *v;
+				struct view *closest_view = NULL;
+				int min_distance = INT_MAX;
+				for_each_view(v, &server->views,
+					LAB_VIEW_CRITERIA_CURRENT_WORKSPACE) {
+					if (v->minimized) {
+						continue;
+					}
+					dx = v->current.x - view->current.x;
+					dy = v->current.y - view->current.y;
+					distance = dx * dx + dy * dy;
+					if ((direction == VIEW_EDGE_UP && dy >= 0)
+						|| (direction == VIEW_EDGE_LEFT && dx >= 0)
+						|| (direction == VIEW_EDGE_DOWN && dy <= 0)
+						|| (direction == VIEW_EDGE_RIGHT && dx <= 0)) {
```

Comment:
Multi-line if statements should be indented too. I suggest following change.
```suggestion
					if ((direction == VIEW_EDGE_UP && dy >= 0)
							|| (direction == VIEW_EDGE_LEFT
								&& dx >= 0)
							|| (direction == VIEW_EDGE_DOWN
								&& dy <= 0)
							|| (direction == VIEW_EDGE_RIGHT
								&& dx <= 0)) {
```